### PR TITLE
NodeMock

### PR DESCRIPTION
**Pull request by ddd-mtl**
## PR summary
Imported and adapted the `TestNode` from `test_bin`. Renamed it `NodeMock.`
Renamed what was the former `integration_test.rs`  to  `engine_test.rs` (no actual change in engine_test.rs)
Now `integration_test.rs` is for testing with NodeMocks.
In this PR, it currently has the same `send_direct_message` test as in engine_test.
You can see how easier it is to write tests with that.

## Review checklist 
- [ ] The story has unit or integration tests
- [ ] No new bugs, and any tech-debt is identified and justified
- [ ] There is enough API documentation (how to use)
- [ ] There is enough code documentation (how the code works)

### CODE REVIEW

File: crates/lib3h/tests/utils/predicate.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+/// Check if Protocol is of type $p
+#[allow(unused_macros)]
+macro_rules! one_is {
```
Comment (zippy) [inline crates/lib3h/tests/utils/predicate.rs]: recommend renaming this from "one_is" to "it_is_a"

File: crates/lib3h/tests/utils/predicate.rs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                return true;
+            }
+            return false;
+        }
+    };
+}
+
+/// Check if Protocol is of type $p and meets conditions set in $code
+#[allow(unused_macros)]
+macro_rules! one_is_where {
```
Comment (zippy) [inline crates/lib3h/tests/utils/predicate.rs]: similarly "it_is_a_where"

File: crates/lib3h/tests/utils/predicate.rs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    ($p:pat, $code:tt) => {
+        move |d| return if let $p = d { $code } else { false }
+    };
+}
+
+#[allow(unused_macros)]
+macro_rules! one_let {
```
Comment (zippy) [inline crates/lib3h/tests/utils/predicate.rs]: is this used anywhere?

Comment (ddd-mtl) [reply crates/lib3h/tests/utils/predicate.rs]: Not yet here. It was used a lot in `test_bin`

**Pull request merged**